ringersong/src/main/java/com/RingerSong/free/data/SpotifyDownloaderRepository.kt

```
@@ -42,66 +44,88 @@ class SpotifyDownloaderRepository(private val context: Context) {
         private const val TAG = "SpotifyDownloader"
     }
 
-    suspend fun downloadTrack(spotifyUrl: String): String? = withContext(Dispatchers.IO) {
-        try {
-            Log.d(TAG, "Starting download for: $spotifyUrl")
+    suspend fun downloadTrack(spotifyUrl: String): DownloadResult = withContext(Dispatchers.IO) {
```

detekt (1.23.8)

[warning] The function downloadTrack appears to be too complex based on Cognitive Complexity (complexity: 34). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)